pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/controllers/controller.go

```diff
@@ -160,6 +174,17 @@ func (r *{{ .Resource.Kind }}Reconciler) Reconcile(ctx context.Context, req ctrl
 	if is{{ .Resource.Kind }}MarkedToBeDeleted {
 		if controllerutil.ContainsFinalizer({{ lower .Resource.Kind }}, {{ lower .Resource.Kind }}Finalizer) {
 			log.Info("Performing Finalizer Operations for {{ .Resource.Kind }} before delete CR")
+
+			// The following implementation will update the status
+			meta.SetStatusCondition(&{{ lower .Resource.Kind }}.Status.Conditions, metav1.Condition{Type: typeDegraded{{ .Resource.Kind }},
+				Status: metav1.ConditionTrue, Reason: "Finalizing",
+				Message: fmt.Sprintf("Performing finalizer operations for the custom resource: %s ", {{ lower .Resource.Kind }}.Name)})
+
+			if err := r.Status().Update(ctx, {{ lower .Resource.Kind }}); err != nil {
+				log.Error(err, "Failed to update {{ .Resource.Kind }} status")
+				return ctrl.Result{}, err
+			}
+
 			r.doFinalizerOperationsFor{{ .Resource.Kind }}({{ lower .Resource.Kind }})
 
```

Comment:
```suggestion
			// TODO(user): If you add operations to the doFinalizerOperationsFor{{ .Resource.Kind }} method then you need to ensure that all worked fine before deleting and updating the Downgrade status
```

Reply:
@NikhilSharmaWe we need after performing the doFrinalizer operations update teh Downgrade status for True, with a message like "Finalize operations for customer resource (%s name) were successfully accomplished"